pkg/reconciler/claimbinding/reconciler.go

```diff
@@ -168,7 +168,11 @@ func defaultCRManaged(m manager.Manager) crManaged {
 			ManagedConfiguratorFn(ConfigureNames),
 			ManagedConfiguratorFn(ConfigureReclaimPolicy),
 		},
-		ManagedCreator:              NewAPIManagedCreator(m.GetClient(), m.GetScheme()),
+		ManagedCreator: NewAPIManagedCreator(m.GetClient(), m.GetScheme()),
+
+		// TODO(negz): Switch to ConnectionPropagator once this has been
+		// deprecated for a release or two.
+		//nolint:staticcheck
 		ManagedConnectionPropagator: resource.NewAPIManagedConnectionPropagator(m.GetClient(), m.GetScheme()),
```

Comment:
Why not use the new one? This is called by `Reconcile` and it's under our control.

Reply:
I was confusing this `managed.ConnectionPublisher`, which is used in a few places outside crossplane-runtime (e.g. in older controllers or controllers that disable publishing via `managed.WithConnectionPublishers()`. I agree that this one is not likely to be used by anyone outside crossplane-runtime.

That said, I don't think it hurts to mark stuff as deprecated for at least one release before we delete it. It's been possible for some time to supply `WithManagedConnectionPropagator` when creating a claim binding controller. I know it's unlikely, but if someone _had_ done this it would be a better experience for this to be deprecated for a release or so before it straight up disappeared. Switching to `ConnectionPropagator` here would mean `WithManagedConnectionPropagator` would no longer work.

Reply:
Yeah I agree with you about marking stuff deprecated before deletion. What I meant here was this line is a usage of the old propagator that we can just replace with the new one. People would import the claim binding reconciler but they don't call its `r.ManagedConnectionPropagator.PropagateConnection()` function, it's the reconciler itself who is calling it. So, the public interface of the reconciler object doesn't change. Same for target reconciler.